File: HW1/gcd.c
/* 
    109971346 Hung-Ruey Chen
    CSE307 Homework 1 
*/
#include <stdio.h>

int main()
{
    int i,j;
    printf("Input num 1:");
    scanf("%d",&i);
    printf("Input num 2:");
    scanf("%d",&j);
    while(i!=j)
    {
        if(i>j) i-=j;
        else j-=i;
    }
    printf("GCD = %d\n",i);
}

File: HW1/gcd.py
def gcd(a,b):
    if a==b:
        return a
    else:
        if a>b:
            return gcd(a-b,a)
        else:
            return gcd(a,b-a)

a=input("Input1:")
b=input("Input2:")
print "GCD = %r" %gcd(a,b)

